hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestFSProvenance.java

```diff
@@ -0,0 +1,301 @@
+/*
+ * Copyright 2019 Apache Software Foundation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hdfs;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.EnumSet;
+
+import io.hops.metadata.hdfs.entity.MetaStatus;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.XAttr;
+import org.apache.hadoop.fs.XAttrSetFlag;
+import org.apache.hadoop.fs.permission.FsPermission;
+import static org.apache.hadoop.hdfs.TestFileCreation.createFile;
+import static org.apache.hadoop.hdfs.TestFileCreation.seed;
+import org.apache.hadoop.hdfs.protocolPB.PBHelper;
+import org.apache.hadoop.hdfs.server.datanode.SimulatedFSDataset;
+import static org.junit.Assert.assertTrue;
+import org.junit.Test;
+
+/**
+ * @author Alex Ormenisan <aaor@kth.se>
+ */
+public class TestFSProvenance {
```

Comment:
I don't really understand what most of these tests are testing. Either they don't have any assert or they have assert on things that are not provenance related.

Reply:
Same here

Reply:
removed the test
It was just a toy for me to test how operations work ... old code.